storage/database/db_manager.go

```diff
@@ -357,11 +374,15 @@ func NewDBManager(dbc *DBConfig) DBManager {
 	} else {
 		checkDBEntryConfigRatio()
 		logger.Info("Partitioned database is used for persistent storage", "DBType", dbc.DBType)
-		if dbm, err := partitionedDatabaseDBManager(dbc); err != nil {
+		dbm, err := partitionedDatabaseDBManager(dbc)
+		if err != nil {
 			logger.Crit("Failed to partitioned database", "DBType", dbc.DBType, "err", err)
-		} else {
-			return dbm
 		}
+		if inMigration, migrationDir := dbm.getStateTrieMigrationInfo(); inMigration {
+			dbm.newStateTrieDB = createStateTrieMigrationDB(dbm.config, migrationDir)
+			dbm.inMigration = true
```

Comment:
Is there any risk of race on those variables?

Reply:
This process will be encapsulated by `Database.Commit` and `Database.Commit` holds `Database.lock` during its operation. Therefore, there seems to be no race for these variables.